--- api/capi.h	2014-02-12 14:13:12.953775117 -0500
+++ /tmp/change/capi.h	2014-02-12 14:02:48.365748031 -0500
@@ -44,8 +44,8 @@
 typedef tesseract::PageSegMode TessPageSegMode;
 typedef tesseract::ImageThresholder TessImageThresholder;
 typedef tesseract::PageIteratorLevel TessPageIteratorLevel;
-typedef tesseract::DictFunc TessDictFunc;
-typedef tesseract::ProbabilityInContextFunc TessProbabilityInContextFunc;
+//typedef tesseract::DictFunc TessDictFunc;
+//typedef tesseract::ProbabilityInContextFunc TessProbabilityInContextFunc;
 // typedef tesseract::ParamsModelClassifyFunc TessParamsModelClassifyFunc;
 typedef tesseract::FillLatticeFunc TessFillLatticeFunc;
 typedef tesseract::Dawg TessDawg;
@@ -263,9 +263,9 @@
 TESS_API BOOL  TESS_CALL TessBaseAPIGetTextDirection(TessBaseAPI* handle, int* out_offset, float* out_slope);
 
 #ifdef TESS_CAPI_INCLUDE_BASEAPI
-TESS_API void  TESS_CALL TessBaseAPISetDictFunc(TessBaseAPI* handle, TessDictFunc f);
+// TESS_API void  TESS_CALL TessBaseAPISetDictFunc(TessBaseAPI* handle, TessDictFunc f);
 TESS_API void  TESS_CALL TessBaseAPIClearPersistentCache(TessBaseAPI* handle);
-TESS_API void  TESS_CALL TessBaseAPISetProbabilityInContextFunc(TessBaseAPI* handle, TessProbabilityInContextFunc f);
+// TESS_API void  TESS_CALL TessBaseAPISetProbabilityInContextFunc(TessBaseAPI* handle, TessProbabilityInContextFunc f);
 
 TESS_API void  TESS_CALL TessBaseAPISetFillLatticeFunc(TessBaseAPI* handle, TessFillLatticeFunc f);
 TESS_API BOOL  TESS_CALL TessBaseAPIDetectOS(TessBaseAPI* handle, OSResults* results);
