--- ../main_dummy.cpp	2012-04-11 20:33:11.103256341 -0400
+++ main_dummy.cpp	2012-04-11 20:33:21.395256463 -0400
@@ -1,4 +1,3 @@
-#include "mfcpch.h"
 // #define USE_VLD //Uncomment for Visual Leak Detector.
 #if (defined _MSC_VER && defined USE_VLD)
 #include <vld.h>
@@ -16,9 +15,8 @@
 #endif
 #include "allheaders.h"
 #include "baseapi.h"
+#include "ctype.h"
 #include "strngs.h"
-#include "tprintf.h"
-#include "tesseractmain.h"
 #ifdef __darwin__
 	#include "fmemopen.h"
 #endif
@@ -159,7 +157,51 @@
 	free(buffer);
 	return retStr;
  }
- 
 
+/*
+	Special function for DiscoveryGarden and OurDigitalWorld
+*/
+int ExtractResultsWrapper(tesseract::TessBaseAPI* api, char *outfile, int chars_limit, char *valid_chars) {
+    char *words;
+    int *lengths;
+    float *costs;
+    int *x0;
+    int *y0;
+    int *x1;
+    int *y1;
+    char *char_4_coords;
+    int *char_x0;
+    int *char_y0;
+    int *char_x1;
+    int *char_y1;
+
+    //allocate space here, we know the number of chars at this point
+    char_4_coords = new char[chars_limit];
+    char_x0 = new int[chars_limit];
+    char_y0 = new int[chars_limit];
+    char_x1 = new int[chars_limit];
+    char_y1 = new int[chars_limit];
+
+    //the all important call
+    api->TesseractExtractResult(&words, &lengths, &costs,
+        &x0, &y0, &x1, &y1,&char_4_coords,&char_x0,&char_y0,
+	&char_x1,&char_y1,api->page_res_, chars_limit);
+
+    FILE *coordfile = NULL;
+    coordfile = fopen(outfile, "w");
+
+    int wordsLen = strlen(char_4_coords);
+    int tmp_cnt = 0;
+
+    for (int i = 0; i < wordsLen; i++) {
+       if (strlen(valid_chars) == 0 || strchr(valid_chars,char_4_coords[i])) {
+          fprintf(coordfile,"%c %d %d %d %d\n",char_4_coords[i],char_x0[i],
+             char_y0[i],char_x1[i],char_y1[i]);
+          tmp_cnt++;
+       }//if
+    }//for
+            
+    fclose(coordfile);
 
-                
+    return tmp_cnt;
+}
