--- main_dummy.cpp.orig	2011-09-18 19:47:40.000000000 -0400
+++ main_dummy.cpp	2011-09-20 09:50:05.000000000 -0400
@@ -16,6 +16,7 @@
 #endif
 #include "allheaders.h"
 #include "baseapi.h"
+#include "ctype.h"
 #include "strngs.h"
 #include "tprintf.h"
 #include "tesseractmain.h"
@@ -159,7 +160,47 @@
 	free(buffer);
 	return retStr;
  }
- 
 
+/*
+	Special function for Discovery Garden and Our Ontario
+*/
+int ExtractResultsWrapper(tesseract::TessBaseAPI* api, char *outfile) {
+    char *words;
+    int *lengths;
+    float *costs;
+    int *x0;
+    int *y0;
+    int *x1;
+    int *y1;
+    char *char_4_coords;
+    int *char_x0;
+    int *char_y0;
+    int *char_x1;
+    int *char_y1;
+
+    //this is lame - but will have to revisit when we figure out what characters JSON trips on
+    const char *valid_characters = " .,abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890!@#^&*()-_+=:;{}[]|?";
+
+
+    //the all important call
+    api->TesseractExtractResult(&words, &lengths, &costs,
+        &x0, &y0, &x1, &y1,&char_4_coords,&char_x0,&char_y0,
+	    &char_x1,&char_y1,api->page_res_);
+
+    FILE *coordfile = NULL;
+    coordfile = fopen(outfile, "w");
+
+    int wordsLen = strlen(char_4_coords);
+
+    int tmpCnt = 0;
+
+    for (int i = 0; i < wordsLen; i++) {
+	    if (strchr(valid_characters,char_4_coords[i]))
+            fprintf(coordfile,"%c %d %d %d %d\n",char_4_coords[i],char_x0[i],char_y0[i],char_x1[i],char_y1[i]);
+    }//for
+            
+    fclose(coordfile);
+
+    return wordsLen;
+}
 
-                
