--- ocr-char-stats.cc.orig	2011-09-18 20:05:00.000000000 -0400
+++ ocr-char-stats.cc	2011-09-18 19:17:36.000000000 -0400
@@ -71,7 +71,7 @@
     }
 
     static void calc_hist(floatarray &hist, floatarray &distances){
-        float maxval=-10000;
+        float maxval=-100000;
         int   len = distances.length();
         for(int i=0; i<len; i++)
             maxval = max(maxval, distances[i]);
@@ -132,8 +132,8 @@
         start = ( (cboxes[0].x0==0) && (cboxes[0].y0==0) ) ? 1 : 0;
         for(int i=start; i< num_chars; i++){
             rectangle r = cboxes[i];
-            minhdist = 10000;
-            minvdist = 10000;
+            minhdist = 100000;
+            minvdist = 100000;
             for(int j=start; j< num_chars; j++){
                 rectangle cand = cboxes[j];
                 if(y_overlap(r,cand) && (cand.x0 > r.x1)){
