--- ocr-deskew-rast.h.orig	2011-09-08 20:28:09.995579186 -0400
+++ ocr-deskew-rast.h	2011-09-02 13:51:47.028079068 -0400
@@ -36,7 +36,7 @@
     struct DeskewPageByRAST : virtual ICleanupBinary, virtual ICleanupGray {
         p_int max_n;
         DeskewPageByRAST() {
-            max_n.bind(this,"max_n",10000,"maximum number of character boxes for deskewing");
+            max_n.bind(this,"max_n",100000,"maximum number of character boxes for deskewing");
         }
         ~DeskewPageByRAST() {
         }
