--- ocr-color-encode-layout.cc.orig	2011-09-08 20:26:23.266208281 -0400
+++ ocr-color-encode-layout.cc	2011-09-02 13:49:49.636661455 -0400
@@ -51,6 +51,9 @@
         }
         color = 1;
         // Limit the number of text columns to what can actually be encoded (30)
+        // Bumped up for newspaper processing (100)
+        int MAX_COLS = 100;
+
         ASSERTWARN(textcolumns.length()<=MAX_COLS);
         int num_cols = (textcolumns.length() < MAX_COLS)?textcolumns.length():MAX_COLS;
         if(num_cols!=textcolumns.length()) {
